tests/subsys/logging/log_core/src/log_core_test.c

```diff
@@ -25,6 +25,9 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 typedef void (*custom_put_callback_t)(struct log_backend const *const backend,
 				      struct log_msg *msg, size_t counter);
+
+static bool in_panic;
```

Comment:
Commit says it's "Add test for message drop notification", but suddenly, "in_panic" is added here.

Reply:
in the commit i've reordered tests to put panic test at the bottom. Unfortunately, there is no cleanup  between test cases (e.g. reset) so logger state stays between tests and there is no going back when panic mode is started. flag in_panic is precaution for the future if someone adds test case after panic test case.

Reply:
ok, let me split that into 2 commits. first tests reordering and then adding new test cases.

Reply:
> ok, let me split that into 2 commits

Only if you find that not too cumbersome. And actually, perfectalize tests is not worth it, better to do it like that next time in such situation.

However, if you'd split commit to logging.h into docstring updates vs drop count support, I'd truly appreciate.

Reply:
log_backend.h cleanup extracted to separate commit.